lib/iris/fileformats/netcdf.py

```diff
@@ -771,7 +772,46 @@ def coord_from_term(term):
         cube.add_aux_factory(factory)
 
 
-def load_cubes(filenames, callback=None):
+def translate_constraints_to_var_callback(constraints):
+    """
+    Translate load constraints into a simple data-var filter function, if possible.
+
+    Returns:
+         * function(cf_var:CFDataVariable): --> bool,
+            or None.
+
+    For now, ONLY handles a single NameConstraint with no 'STASH' component.
+
+    """
+    constraints = iris._constraints.list_of_constraints(constraints)
+    result = None
+    if len(constraints) == 1:
+        (constraint,) = constraints
+        if (
+            isinstance(constraint, iris._constraints.NameConstraint)
+            and constraint.STASH == "none"
+        ):
+            # As long as it doesn't use a STASH match, then we can treat it as
+            # a testing against name properties of cf_var.
+            # That's just like testing against name properties of a cube, except that they may not all exist.
+            def inner(cf_datavar):
+                match = True
+                for name in constraint._names:
+                    expected = getattr(constraint, name)
+                    if name != "STASH" and expected != "none":
+                        # Fetch property : N.B. CFVariable caches the property values
+                        # The use of a default here is the only difference from the code in NameConstraint.
+                        actual = getattr(cf_datavar, name, "")
+                        if actual != expected:
```

Comment:
N.B. this still works if 'expected' happened to be a function or a list of values, instead of a simple constant.
Which must be considered as [those are also valid possibilties](https://scitools-iris.readthedocs.io/en/latest/generated/api/iris.html#iris.NameConstraint)

Reply:
Looking again, no longer convinced this is 'real' problem.  A function will not compare to string, so no problem ??